### PR TITLE
SciPy 2019 submission

### LINKED ISSUE #396: Zarr at Scipy 2019
I think someone should propose a talk about Zarr to Scipy. This would really help raise the profile of the project to a broad audience.

Deadline is Feb. 11:
https://www.scipy2019.scipy.org/talk-poster-presentations

### PR DESCRIPTION
**Pull request by alimanfoo**
This PR has a draft submission for a talk on Zarr for the SciPy 2019 conference. 

Comments and suggestions very welcome.

Resolves #396.

### CODE REVIEW

File: docs/talks/scipy2019/submission.rst [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+[1_], and have released an open source Python implementation [2_,
+3_]. Our approach builds on data storage concepts from HDF5 [4_],
+particularly chunking and compression, and hierarchical organisation
+of datasets. Key design goals include: platform-independent and as
+simple as possible to implement in other programming languages;
+support for multiple concurrent readers or writers; support for a
+variety of parallel computing environments, from multi-threaded
+execution on a single CPU to multi-process execution across a
+multi-node cluster; pluggable storage subsystem with support for cloud
+object stores; pluggable encoding subsystem with support for a variety
```
Comment (jhamman) [inline docs/talks/scipy2019/submission.rst]: suggest: pluggable storage subsystem with support for cloud object stores, file systems, or key-value databases

File: docs/talks/scipy2019/submission.rst [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+Conclusions
+~~~~~~~~~~~
+
+Zarr is generating interest from potential users across a wide range
+of scientific domains, and work is ongoing to establish a community
+process for further development of the specifications and
+implementations in other programming languages. Zarr has recently been
+presented in several venues, including a webinar for the ESIP
+Federation tech dive series [8_], and @@TODO @rabernat's recent talk
+[9_].
```
Comment (jhamman) [inline docs/talks/scipy2019/submission.rst]: @rabernat's talk was at AGU, link: https://agu.confex.com/agu/fm18/meetingapp.cgi/Paper/390015

File: docs/talks/scipy2019/submission.rst [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+Results
+~~~~~~~
+
+We illustrate the use of Zarr with examples from several different
+scientific domains. Zarr is being used within the Pangeo project [5_],
+which is building a community platform for big data geoscience. The
+Pangeo community have converted several existing datasets to Zarr,
+including @@TODO, and have demonstrated their use in computations
+including @@TODO. Within the MalariaGEN project [6_], Zarr is used to
```
Comment (jhamman) [inline docs/talks/scipy2019/submission.rst]: I think you could just say something like:

```
The Pangeo community has converted several existing climate modeling and satellite observation datasets to Zarr, and has demonstrated their use in computations using HPC and Cloud computing environments.
```

The list of zarr datasets in Pangeo is too long to enumerate at this point. Partial list here: http://pangeo.io/catalog.html

**Comment by alimanfoo** [comment]
Thanks @jhamman :+1:

File: docs/talks/scipy2019/submission.rst (hunk 1 above)
Comment (alimanfoo) [reply docs/talks/scipy2019/submission.rst]: addressed

File: docs/talks/scipy2019/submission.rst (hunk 2 above)
Comment (alimanfoo) [reply docs/talks/scipy2019/submission.rst]: addressed

File: docs/talks/scipy2019/submission.rst (hunk 3 above)
Comment (alimanfoo) [reply docs/talks/scipy2019/submission.rst]: addressed

**Comment by alimanfoo** [comment]
Thanks again @jhamman, I've added your suggestions.

File: docs/talks/scipy2019/submission.rst [hunk 4]
```diff
@@ -85,11 +101,41 @@ References
...
 
 Authors
 -------
 
-@@TODO
+Project contributors are listed in alphabetical order by surname.
+
+* `Alistair Miles <https://github.com/alimanfoo>`_, University of Oxford
+* `John Kirkham <https://github.com/jakirkham>`_, HHMI Janelia
+* `Ryan Abernathy <https://github.com/rabernat>`_, Columbia University
```
Comment (rabernat) [inline docs/talks/scipy2019/submission.rst]: Abernathey

File: docs/talks/scipy2019/submission.rst [hunk 5]
```diff
@@ -49,29 +58,36 @@ encoding subsystem with support for a variety of modern compressors.
...
-We illustrate the use of Zarr with examples from several different
-scientific domains. Zarr is being used within the Pangeo project [5_],
-which is building a community platform for big data geoscience. The
-Pangeo community have converted a number of existing climate modelling
-and satellite observation datasets to Zarr [6_], and has demonstrated
+We illustrate the use of Zarr with examples from several scientific
+domains. Zarr is being used within the Pangeo project [5_], which is
+building a community platform for big data geoscience. The Pangeo
+community have converted a number of existing climate modelling and
+satellite observation datasets to Zarr [6_], and has demonstrated
```
Comment (rabernat) [inline docs/talks/scipy2019/submission.rst]: *have* demonstrated

File: docs/talks/scipy2019/submission.rst (hunk 4 above)
Comment (alimanfoo) [reply docs/talks/scipy2019/submission.rst]: Sorry! Fixed.

**Comment by alimanfoo** [comment]
Thanks @rabernat.

**Comment by jeromekelleher** [comment]
LGTM, great work all!

File: docs/talks/scipy2019/submission.rst [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+* https://github.com/dazzag24
+* `Martin Durant <https://github.com/martindurant>`_, Anaconda, inc.
+* `Jan Funke <https://github.com/funkey>`_, HHMI Janelia
+* `Fabian Gans <https://github.com/meggart>`_, Max Planck Institute
+* `Joe Hamman <https://github.com/jhamman>`_, NCAR
+* `Stephan Hoyer <https://github.com/shoyer`_, Google
+* `Jerome Kelleher <https://github.com/jeromekelleher>`_, University of Oxford
+* `John Kirkham <https://github.com/jakirkham>`_, HHMI Janelia
+* `Alistair Miles <https://github.com/alimanfoo>`_, University of Oxford
+* `Josh Moore <https://github.com/joshmoore>`_, Glencoe Software
```
Comment (joshmoore) [inline docs/talks/scipy2019/submission.rst]:
```suggestion
* `Josh Moore <https://github.com/joshmoore>`_, University of Dundee
```

File: docs/talks/scipy2019/submission.rst [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+* `Charles Noyes <https://github.com/CSNoyes>`_, University of Southern California
+* `Tarik Onalan <https://github.com/onalant>`_
+* `Constantin Pape <https://github.com/constantinpape>`_, University of Heidelberg
+* `Zain Patel <https://github.com/mzjp2>`_, University of Cambridge
+* `Matthew Rocklin <https://github.com/mrocklin>`_, NVIDIA
+* `Stephan Saafeld <https://github.com/axtimwalde>`_, HHMI Janelia
+* `Vincent Schut <https://github.com/vincentschut>`_, Satelligence
+* https://github.com/shikharsg
```
Comment (shikharsg) [inline docs/talks/scipy2019/submission.rst]: * `Shikhar Goenka <https://github.com/shikharsg>`_, Satavia

File: docs/talks/scipy2019/submission.rst [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+Authors
+-------
+
+Project contributors are listed in alphabetical order by surname.
+
+* `Ryan Abernathey <https://github.com/rabernat>`_, Columbia University 
+* `Stephan Balmer <https://github.com/sbalmer>`_
+* `Ambrose Carr <https://github.com/ambrosejcarr>`_, Chan Zuckerberg Initiative
+* `Tim Crone <https://github.com/tjcrone>`_, Columbia University
+* https://github.com/dazzag24
```
Comment (shikharsg) [inline docs/talks/scipy2019/submission.rst]: * `Darren Gallagher <https://github.com/dazzag24>`_, Satavia

File: docs/talks/scipy2019/submission.rst [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+on a variety of platforms.
+
+Methods
+~~~~~~~
+
+We have designed a new storage format and protocol for tensor data
+[1_], and have released an open source Python implementation [2_,
+3_]. Our approach builds on data storage concepts from HDF5 [4_],
+particularly chunking and compression, and hierarchical organisation
+of datasets. Key design goals include: platform-independent and as
```
Comment (jakirkham) [inline docs/talks/scipy2019/submission.rst]: Maybe we should clarify what is "platform-independent". IOW is the implementation intended to be platform independent or the storage format itself? Also not sure if we want the word platform-independent or platform-agnostic.

File: docs/talks/scipy2019/submission.rst [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+simple as possible to implement in other programming languages;
```
Comment (jakirkham) [inline docs/talks/scipy2019/submission.rst]: Might shorten this to just "simple implementation", "simple design spec", or similar.

File: docs/talks/scipy2019/submission.rst [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+.. _15: https://agu.confex.com/agu/fm18/meetingapp.cgi/Paper/390015
+
+
+Authors
+-------
+
+Project contributors are listed in alphabetical order by surname.
+
+* `Ryan Abernathey <https://github.com/rabernat>`_, Columbia University 
+* `Stephan Balmer <https://github.com/sbalmer>`_
```
Comment (jakirkham) [inline docs/talks/scipy2019/submission.rst]: Meteotest?

cc @sbalmer

File: docs/talks/scipy2019/submission.rst [hunk 12]
```diff
@@ -0,0 +1,141 @@
...
+* `Jan Funke <https://github.com/funkey>`_, HHMI Janelia
+* `Fabian Gans <https://github.com/meggart>`_, Max Planck Institute
+* `Joe Hamman <https://github.com/jhamman>`_, NCAR
+* `Stephan Hoyer <https://github.com/shoyer`_, Google
+* `Jerome Kelleher <https://github.com/jeromekelleher>`_, University of Oxford
+* `John Kirkham <https://github.com/jakirkham>`_, HHMI Janelia
+* `Alistair Miles <https://github.com/alimanfoo>`_, University of Oxford
+* `Josh Moore <https://github.com/joshmoore>`_, University of Dundee
+* `Charles Noyes <https://github.com/CSNoyes>`_, University of Southern California
+* `Tarik Onalan <https://github.com/onalant>`_
```
Comment (jakirkham) [inline docs/talks/scipy2019/submission.rst]: Is there an org you would like to list here, @onalant?

**Comment by jakirkham** [comment]
Thanks @alimanfoo for working on this. Added a few comments above. Looks pretty good.

**Comment by jakirkham** [comment]
Oh, sorry, one other question, should we list parallel libraries that Zarr works with? For instance both Dask and Xarray are able to read and write Zarr. Also it sounds like users have had success working with Zarr on Spark and MPI. Maybe there are others I've missed?

**Comment by jakirkham** [comment]
Just as an FYI, for the author list when submitting this, they will want first (optional) & last name, email, country, organization, and (optional) webpage for each author.

File: docs/talks/scipy2019/submission.rst (hunk 7 above)
Comment (shikharsg) [reply docs/talks/scipy2019/submission.rst]: @alimanfoo

File: docs/talks/scipy2019/submission.rst (hunk 8 above)
Comment (shikharsg) [reply docs/talks/scipy2019/submission.rst]: @alimanfoo

File: docs/talks/scipy2019/submission.rst (hunk 7 above)
Comment (alimanfoo) [reply docs/talks/scipy2019/submission.rst]: Thanks @shikharsg, will fix.

File: docs/talks/scipy2019/submission.rst (hunk 8 above)
Comment (alimanfoo) [reply docs/talks/scipy2019/submission.rst]: Thanks again.

**Comment by alimanfoo** [comment]
Thanks @jakirkham, I've pushed some changes which incorporate your suggestions.

File: docs/talks/scipy2019/submission.rst [hunk 13]
```diff
@@ -0,0 +1,144 @@
...
+* `Stephan Balmer <https://github.com/sbalmer>`_, Meteotest
+* `Ambrose Carr <https://github.com/ambrosejcarr>`_, Chan Zuckerberg Initiative
+* `Tim Crone <https://github.com/tjcrone>`_, Columbia University
+* `Martin Durant <https://github.com/martindurant>`_, Anaconda, inc.
+* `Jan Funke <https://github.com/funkey>`_, HHMI Janelia
+* `Darren Gallagher <https://github.com/dazzag24>`_, Satavia
+* `Fabian Gans <https://github.com/meggart>`_, Max Planck Institute
+* `Shikhar Goenka <https://github.com/shikharsg>`_, Satavia
+* `Joe Hamman <https://github.com/jhamman>`_, NCAR
+* `Stephan Hoyer <https://github.com/shoyer`_, Google
```
Comment (jakirkham) [inline docs/talks/scipy2019/submission.rst]:
```suggestion
* `Stephan Hoyer <https://github.com/shoyer>`_, Google
```

**Comment by jakirkham** [comment]
Thanks @alimanfoo. Looks good to me. Caught a small typo referenced above.

**Comment by alimanfoo** [comment]
Alright, submission done! Thanks everyone.
